Watch unit test flake on Travis OS X

```
panic: Fail in goroutine after TestWatch/file_delete has completed
goroutine 37 [running]:
testing.(*common).Fail(0xc420428780)
	/Users/travis/.gimme/versions/go1.10.4.darwin.amd64/src/testing/testing.go:509 +0x134
testing.(*common).Error(0xc420428780, 0xc420055f40, 0x1, 0x1)
	/Users/travis/.gimme/versions/go1.10.4.darwin.amd64/src/testing/testing.go:579 +0x6f
github.com/GoogleContainerTools/skaffold/testutil.CheckError(0xc420428780, 0x1731800, 0x1728ec0, 0xc42000c1a0)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/testutil/util.go:83 +0xa3
github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.TestWatch.func7.1(0x172b540, 0xc4202426a0, 0x1731840, 0xc42009c740, 0xc4200aa0e8, 0xc4202b22e0, 0xc420428780)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/watch_test.go:81 +0xd5
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.TestWatch.func7
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/watch_test.go:79 +0x296
FAIL	github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch	0.134s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/yamltags	0.008s	coverage: 94.0% of statements
?   	github.com/GoogleContainerTools/skaffold/testutil	[no test files]
make: *** [coverage] Error 1
```

https://travis-ci.org/GoogleContainerTools/skaffold/jobs/427999539

--- a/pkg/skaffold/watch/changes.go
+++ b/pkg/skaffold/watch/changes.go
@@ -51,6 +51,10 @@ func lastModified(paths []string) (time.Time, error) {
 	for _, path := range paths {
 		stat, err := os.Stat(path)
 		if err != nil {
+			if os.IsNotExist(err) {
+				continue // Ignore files that don't exist
+			}
+
 			return last, errors.Wrapf(err, "unable to stat file %s", path)
 		}
 

--- a/pkg/skaffold/watch/watch_test.go
+++ b/pkg/skaffold/watch/watch_test.go
@@ -76,8 +76,11 @@ func TestWatch(t *testing.T) {
 
 			// Run the watcher
 			ctx, cancel := context.WithCancel(context.Background())
+			var stopped sync.WaitGroup
+			stopped.Add(1)
 			go func() {
 				err = watcher.Run(ctx, 10*time.Millisecond, somethingChanged.callNoErr)
+				stopped.Done()
 				testutil.CheckError(t, false, err)
 			}()
 
@@ -87,6 +90,7 @@ func TestWatch(t *testing.T) {
 			folderChanged.wait()
 			somethingChanged.wait()
 			cancel()
+			stopped.Wait() // Make sure the watcher is stopped before deleting the tmp folder
 		})
 	}
 }

